src/process/ProcessManagerImpl.cpp

```diff
@@ -70,30 +72,40 @@ class ProcessExitEvent::Impl
     Impl(std::shared_ptr<RealTimer> const& outerTimer,
          std::shared_ptr<asio::error_code> const& outerEc,
          std::string const& cmdLine, std::string const& outFile,
-         std::weak_ptr<ProcessManagerImpl> pm)
+         std::string const& tempFile, std::weak_ptr<ProcessManagerImpl> pm)
         : mOuterTimer(outerTimer)
         , mOuterEc(outerEc)
         , mCmdLine(cmdLine)
         , mOutFile(outFile)
-        , mTempFile(binToHex(randomBytes(5)))
+        , mTempFile(tempFile)
 #ifdef _WIN32
         , mProcessHandle(outerTimer->get_executor())
 #endif
         , mProcManagerImpl(pm)
     {
     }
 
-    void
+    bool
     finish()
     {
-        if (!mOutFile.empty() && mRunning)
+        if (!mOutFile.empty())
         {
-            if (std::rename(mTempFile.c_str(), mOutFile.c_str()))
+            if (fs::exists(mOutFile))
+            {
+                CLOG(WARNING, "Process")
+                    << "Outfile already exists: " << mOutFile;
+                return false;
```

Comment:
should this function return an `ec` instead of a `bool`?

Reply:
I was saying that so that you can return different error codes based on the type of error instead of performing a blanket conversion

Reply:
ok let me update it